From 96e5d0bd940fcbecdc93a24f8a9f0c568519c254 Mon Sep 17 00:00:00 2001
From: Dave Kleikamp <dave.kleikamp@oracle.com>
Date: Wed, 14 Dec 2011 12:22:44 -0800
Subject: [PATCH] Asynchronous I/O latency to a solid-state disk greatly
 increased between the 2.6.32 and 3.0 kernels. By removing
 the plug from do_io_submit(), we observed a 34% improvement
 in the I/O latency.

Unfortunately, at this level, we don't know if the request is to
a rotating disk or not.

Signed-off-by: Dave Kleikamp <dave.kleikamp@oracle.com>
Cc: linux-aio@kvack.org
Cc: Chris Mason <chris.mason@oracle.com>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: Andi Kleen <ak@linux.intel.com>
Cc: Jeff Moyer <jmoyer@redhat.com>
---
 fs/aio.c |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/fs/aio.c b/fs/aio.c
index e29ec48..75e05c9 100644
--- a/fs/aio.c
+++ b/fs/aio.c
@@ -1622,7 +1622,6 @@ long do_io_submit(aio_context_t ctx_id, long nr,
 	struct kioctx *ctx;
 	long ret = 0;
 	int i;
-	struct blk_plug plug;
 
 	if (unlikely(nr < 0))
 		return -EINVAL;
@@ -1639,8 +1638,6 @@ long do_io_submit(aio_context_t ctx_id, long nr,
 		return -EINVAL;
 	}
 
-	blk_start_plug(&plug);
-
 	/*
 	 * AKPM: should this return a partial result if some of the IOs were
 	 * successfully submitted?
@@ -1663,7 +1660,6 @@ long do_io_submit(aio_context_t ctx_id, long nr,
 		if (ret)
 			break;
 	}
-	blk_finish_plug(&plug);
 
 	put_ioctx(ctx);
 	return i ? i : ret;
-- 
1.7.10
